notebooks/228-clip-zero-shot-image-classification/228-clip-quantize-nncf.ipynb

```diff
@@ -0,0 +1,410 @@
+    "\n",
+    "def get_pil_from_url(url):\n",
+    "    \"\"\"\n",
+    "    Downloads and converts an image from a URL to a PIL Image object.\n",
+    "    \"\"\"\n",
+    "    response = requests.get(url, verify=False, timeout=20)\n",
+    "    image = Image.open(BytesIO(response.content))\n",
+    "    return image.convert(\"RGB\")\n",
+    "\n",
+    "def collate_fn(example, image_column=\"url\", text_column=\"caption\"):\n",
+    "    \"\"\"\n",
+    "    Preprocesses an example by loading and transforming image and text data.\n",
+    "    Checks if the text data in the example is valid by calling the `check_text_data` function.\n",
+    "    Downloads the image specified by the URL in the image_column by calling the `get_pil_from_url` function.\n",
+    "    If there is any error during the download process, returns None.\n",
+    "    Returns the preprocessed inputs with transformed image and text data.\n",
+    "    \"\"\"\n",
+    "    assert len(example) == 1\n",
+    "    example = example[0]\n",
+    "\n",
+    "    if not check_text_data(example[text_column]):\n",
+    "        raise ValueError(\"Text data is not valid\")\n",
+    "\n",
+    "    url = example[image_column]\n",
+    "    try:\n",
+    "        image = get_pil_from_url(url)\n",
+    "    except Exception:\n",
+    "        return None\n",
+    "\n",
+    "    inputs = processor(text=example[text_column], images=[image], return_tensors=\"pt\", padding=True)\n",
+    "    if inputs['input_ids'].shape[1] > max_length:\n",
+    "        return None\n",
+    "    return inputs"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "befc2204",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import torch\n",
+    "from datasets import load_dataset\n",
+    "\n",
+    "def prepare_calibration_data(dataloader, init_steps):\n",
+    "    \"\"\"\n",
+    "    This function prepares calibration data from a dataloader for a specified number of initialization steps.\n",
+    "    It iterates over the dataloader, fetching batches and storing the relevant data.\n",
+    "    \"\"\"\n",
+    "    data = []\n",
+    "    print(f\"Fetching {init_steps} for the initialization...\")\n",
+    "    counter = 0\n",
+    "    for batch in dataloader:\n",
+    "        if counter == init_steps:\n",
+    "            break\n",
+    "        if batch:\n",
+    "            counter += 1\n",
+    "            with torch.no_grad():\n",
+    "                data.append(\n",
+    "                    {\n",
+    "                        \"pixel_values\": batch[\"pixel_values\"].to(\"cpu\"),\n",
+    "                        \"input_ids\": batch[\"input_ids\"].to(\"cpu\"),\n",
+    "                        \"attention_mask\": batch[\"attention_mask\"].to(\"cpu\")\n",
+    "                    }\n",
+    "                )\n",
+    "    return data\n",
+    "\n",
+    "\n",
+    "def prepare_dataset(opt_init_steps=300, max_train_samples=1000):\n",
```

Comment:
I get an error in this step (suppose it's due to Intel politics and the dataset is not loaded):

RuntimeError: DataLoader worker (pid(s) 11696, 16624) exited unexpectedly

Reply:
Removed `num_workers=2` from `torch.utils.data.DataLoader`